File: day02_pt2.js
// a rock 1
// b paper 2
// c scissors 3
// x lose
// y draw
// z win

const input = require("fs")
  .readFileSync("day02_input.txt")
  .toString()
  .split(/\r?\n/)
//console.log(input)
let answer = 0
for (let i = 0; i < input.length; i++) {
  let [opp, me] = input[i].split(" ")
  answer += calculateScore(opp, me)
}

console.log({ answer })
function calculateScore(opp, me) {
  if (opp === "A") {
    // rock
    switch (me) {
      case "X":
        // play sci
        return 3
        break
      case "Y":
        //play rock
        return 4
        break
      case "Z":
        // play paper
        return 8
    }
  } else if (opp === "B") {
    // paper
    switch (me) {
      case "X":
        //rock
        return 1
        break
      case "Y":
        //paper
        return 5
        break
      case "Z":
        //sci
        return 9
    }
  } else if ((opp = "C")) {
    // sci
    switch (me) {
      case "X":
        //paper
        return 2
        break
      case "Y":
        //sci
        return 6
        break
      case "Z":
        //rock
        return 7
    }
  } else {
    console.log(`invalid input ${opp}, ${me}`)
  }
}

File: day02_pt2.py
with open('day02_input.txt') as f:
    matches = f.read().split("\n")


def calculateScore(opp, me):
    if opp == "A":
        if me == "X":
            return 3
        elif me == "Y":
            return 4
        elif me == "Z":
            return 8
    elif opp == "B":
        if me == "X":
            return 1
        elif me == "Y":
            return 5
        elif me == "Z":
            return 9
    elif opp == "C":
        if me == "X":
            return 2
        elif me == "Y":
            return 6
        elif me == "Z":
            return 7


answer = 0
for match in matches:
    opp, me = match.split(" ")
    answer += calculateScore(opp, me)
print(answer)
